### ISSUE TITLE
prompt does not find external commands

### ISSUE THREAD

**Issue by Fishrock123** [nushell/nushell#3134]
**Describe the bug**

Config:
```toml
prompt = "STARSHIP_SHELL= starship prompt"
```

Output:
```
Welcome to Nushell 0.27.1 (type 'help' for more info)
'starship' is not recognized as an internal or external command,
operable program or batch file.
```

```
> which starship
───┬──────────┬───────────────────────────────────────────┬─────────
 # │   arg    │                   path                    │ builtin 
───┼──────────┼───────────────────────────────────────────┼─────────
 0 │ starship │ C:\Users\Jeremiah\.cargo\bin\starship.exe │ false   
───┴──────────┴───────────────────────────────────────────┴─────────
```

**Configuration (please complete the following information):**

Run `version | pivot` and paste the output to show OS, features, etc.

```
❯ version | pivot
───┬────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────
 # │      Column0       │                                                 Column1
───┼────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────
 0 │ version            │ 0.27.1
 1 │ build_os           │ windows-x86_64
 2 │ rust_version       │ rustc 1.50.0 (cb75ad5db 2021-02-10)
 3 │ rust_channel       │ stable-x86_64-pc-windows-msvc (default)
 4 │ cargo_version      │ cargo 1.50.0 (f04e7fab7 2021-02-04)
 5 │ pkg_version        │ 0.27.1
 6 │ build_time         │ 2021-03-06 03:15:01
 7 │ build_rust_channel │ release
 8 │ features           │ clipboard-cli, ctrlc, default, directories, dirs, ichwh, ptree, rustyline, term, trash, uuid, which, zip
───┴────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Add any other context about the problem here.**
reported in discord https://discord.com/channels/601130461678272522/614593951969574961/817621571468394546

**Comment by Fishrock123** [comment]
This seems to only happen on the first couple invocations, at some point it starts finding it.

**Comment by Type1J** [comment]
I can confirm:
After a `` config set prompt `echo $(starship prompt)`  ``, I get the following when starting `nu`:
```
Welcome to Nushell 0.27.1 (type 'help' for more info)
error: Failed to spawn process
  ┌─ shell:1:8
  │
1 │ echo $(starship prompt)
  │        ^^^^^^^^ failed to spawn
  ```
If I press enter, it repeats the error. However, if I run any app that requires searching the path once manually (even `starship prompt` itself) , the prompt starts working normally.

**Comment by ahkrr** [comment]
I looked into this and made a PR that fixes this at #3189

**Comment by Fishrock123** [comment]
This was fixed
